int32 saturation
---
int32 error